### ISSUE TITLE
terminate called after throwing an instance of 'std::bad_weak_ptr' 

### ISSUE THREAD

**Issue by zlw21gxy** [carla-simulator/carla#1399]
![image](https://user-images.githubusercontent.com/46110602/54472904-694ced00-480b-11e9-9897-9493da37bd7d.png)

hey guys, I am using Carla 0.9.4 and Ubuntu 16.04
does anyone why this error happen? thanks!!!!
I guess it's due to listening to the sensor in the wrong way

**Comment by nsubiron** [comment]
Hi @zlw21gxy, do you have some minimal code example reproducing this issue?

**Comment by zlw21gxy** [comment]
thanks for help and sorry for late reply, this days I am trying to locate the error
![image](https://user-images.githubusercontent.com/46110602/55141842-40ffaf80-5176-11e9-9fb3-c46192f60eb1.png)
it is caused by spawn camera in world, but this error won't happened at the beginning, it only happened when I running program for a while, like couple of hours. @nsubiron

**Comment by zlw21gxy** [comment]
#1460

**Comment by fpasch** [comment]
I'm facing a similar issue when using the ros-bridge (and starting/stopping another client).
I believe the issue is related to having two clients accessing the data.

My clients:
1. ros-bridge
2. manual_control.py

You can reproduce it like that.

1. execute ros-bridge
2. Restart manual_control.py every 4 seconds (e.g with the help of this script)
```
#!/usr/bin/env python
import signal
import subprocess
import time

ret = 0
while ret is 0:
    pro = subprocess.Popen("./manual_control.py")
    time.sleep(4)
    pro.terminate()
    pro.wait()
    ret = pro.returncode
print("returned {}".format(ret))
```

Afer a few restarts of manual-control, the ROS bridge will report the error and stops.

**Comment by zdx3578** [comment]
@nsubiron

**Comment by nsubiron** [comment]
It looks like it's a bug inside LibCarla, related to the client objects used to subscribe to a sensor. I'll try to create a minimal test reproducing this in C++. Unfortunately we won't be able to fix it for 0.9.5.

**Comment by oldtopos** [comment]
When runnng ros_bridge, I commented line 167 of parent.py

self.child_actors[actor_id].destroy()

This has greatly improved ros_bridge stability; I expect it is causing a double delete attempt in the python API code ...

**Comment by fpasch** [comment]
@oldtopos What you're describing is a different issue, which is, as you already stated ros-bridge only. I've create an issue in ros-bridge > https://github.com/carla-simulator/ros-bridge/issues/133

**Comment by germanros1987** [comment]
This should be solved now in 0.9.8.
